### PR TITLE
Modify pass pipeline to allow lowering `tt.load` to 2DBlockRead

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

File: lib/Dialect/TritonGPU/Transforms/Coalesce.cpp [hunk 1]
```diff
@@ -164,13 +164,11 @@ struct CoalescePass : public TritonGPUCoalesceBase<CoalescePass> {
       Value ptr = getMemAccessPtr(curr);
       if (!ptr)
         return;
-      // We only convert `tensor<tt.ptr<>>` or `tt.ptr<tensor<>>` load/store
-      bool isPtrTensor = false, isTensorPointer = false;
+      // We only convert `tensor<tt.ptr<>>` load/store
```
Comment (whitneywhtsang) [inline lib/Dialect/TritonGPU/Transforms/Coalesce.cpp:167]: Should we upstream this change?

Comment (etiotto) [reply lib/Dialect/TritonGPU/Transforms/Coalesce.cpp:167]: Poste a PR upstream: https://github.com/openai/triton/pull/3866

**Comment by etiotto** [comment]
Now, with this PR, the code generated for `09-experimental-block-pointer.py` shows that the 2D block read operation is used:
```
TRITON_DISABLE_LINE_INFO=1  MLIR_ENABLE_DUMP=1  python 09-experimental-block-pointer.py >! log 2>&1 &  
grep "BlockRead" log| wc                                                                                                      ✔  10361  17:46:33 
    712   22416  158600
```

**Comment by etiotto** [comment]
@whitneywhtsang @chengjunlu I'd like to land this PR asap. If there are concerns with relaxing the precision of the tests for flash attention we can open a separate defect to track that.

**Review by whitneywhtsang — APPROVED** [review]

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h:1]: These DotOpLayout changes should be upstreamed as well.

**Pull request merged**